pkgs/desktops/gnome/apps/gnome-text-editor/default.nix

```diff
@@ -0,0 +1,76 @@
+{lib
+, stdenv
+, meson
+, fetchurl
+, python3
+, pkg-config
+, gtk4
+, glib
+, gtksourceview5
+, gsettings-desktop-schemas
+, wrapGAppsHook
+, ninja
+, gnome
+, enchant
+, icu
+, itstool
+, libadwaita
+, libxml2
+, pcre
+, appstream-glib
+, desktop-file-utils
+}:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-text-editor";
+  version = "41.1";
+
+  src = fetchurl {
+    url = "mirror://gnome/sources/gnome-text-editor/${lib.versions.major version}/${pname}-${version}.tar.xz";
+    sha256 = "sha256-YZ7FINbgkF1DEWcCTkPc4Nv2o0Xy1IaTUB1w3HYm+GE=";
+  };
+
+  nativeBuildInputs = [
+    appstream-glib
+    desktop-file-utils
+    itstool
+    libxml2 # for xmllint
+    meson
+    ninja
+    pkg-config
+    python3
+    wrapGAppsHook
```

Comment:
Maybe we can use `wrapGAppsHook4`? We can do `substituteInPlace build-aux/meson/postinstall.py --replace "gtk-update-icon-cache" "gtk4-update-icon-cache"`, though I noticed that the file has been [dropped](https://gitlab.gnome.org/GNOME/gnome-text-editor/-/commit/0a9bca3e6460435ca8a2c96e4a6af7ef920465d4) in `main`.